### PR TITLE
Use an enum for configurable_alternatives to make the generated json schema nicer

### PR DESCRIPTION
**Pull request by nfcampos**

### CODE REVIEW

**Review by eyurtsev — APPROVED** [review]

File: libs/langchain/langchain/schema/runnable/base.py [hunk 1]
```diff
@@ -136,6 +136,7 @@ class _Config:
             arbitrary_types_allowed = True
 
         include = include or []
+        include = [i for i in include if i != "configurable"]
```
Comment (eyurtsev) [inline libs/langchain/langchain/schema/runnable/base.py:139]: could we make it possible to generate schema that does not include "configurable" data?

Comment (nfcampos) [reply libs/langchain/langchain/schema/runnable/base.py:139]: how is that useful?

Comment (nfcampos) [reply libs/langchain/langchain/schema/runnable/base.py:139]: opt in to expose configuration params twice seems overkill no?

File: libs/langchain/langchain/schema/runnable/configurable.py [hunk 2]
```diff
@@ -239,16 +239,15 @@ class RunnableConfigurableAlternatives(DynamicRunnable[Input, Output]):
 
     @property
     def config_specs(self) -> Sequence[ConfigurableFieldSpec]:
-        alt_keys = self.alternatives.keys()
-        which_keys = tuple(Literal[k] for k in alt_keys) + (  # type: ignore
-            Literal["default"],
+        which_enum = enum.StrEnum(  # type: ignore[misc]
+            self.which.name, list(self.alternatives.keys()) + ["default"]
```
Comment (obi1kenobi) [inline libs/langchain/langchain/schema/runnable/configurable.py:243]: Should we worry about this note from the docs on `StrEnum`?
image

File: libs/langchain/langchain/schema/runnable/configurable.py [hunk 3]
```diff
@@ -239,16 +239,15 @@ class RunnableConfigurableAlternatives(DynamicRunnable[Input, Output]):
 
     @property
     def config_specs(self) -> Sequence[ConfigurableFieldSpec]:
-        alt_keys = self.alternatives.keys()
-        which_keys = tuple(Literal[k] for k in alt_keys) + (  # type: ignore
-            Literal["default"],
+        which_enum = enum.StrEnum(  # type: ignore[misc]
```
Comment (obi1kenobi) [inline libs/langchain/langchain/schema/runnable/configurable.py:242]: Our of curiosity, what is the error this is suppressing?

Comment (nfcampos) [reply libs/langchain/langchain/schema/runnable/configurable.py]: Pylance/mypy doesn't like a non-constant value being passed in as the name of the enum, but from checking the code this doesnt raise an issue

File: libs/langchain/langchain/schema/runnable/configurable.py (hunk 2 above)
Comment (nfcampos) [reply libs/langchain/langchain/schema/runnable/configurable.py]: It's fine, we only check these values in this component

File: libs/langchain/langchain/schema/runnable/base.py (hunk 1 above)
Comment (eyurtsev) [reply libs/langchain/langchain/schema/runnable/base.py:139]: * either we do that or that key is treated differently from other top level keys of the RunnableConfig (which is ifne)
* Any situations where a server picks up an existing configurable runnable without knowing that it's configurable and not wanting to expose configurability by default?

**Review by eyurtsev — APPROVED** [review]

**Pull request merged**